py/torch_tensorrt/dynamo/conversion/impl/select.py

```diff
@@ -47,26 +47,16 @@ def select(
     if dynamic_shape:
         # Check whether slice target dim is dynamic shape dim
         assert input.shape[dim] != -1, "Can't select on negative shape dimension!"
-    index = index
 
     if index >= input.shape[dim]:
         raise RuntimeError(
             f"cannot have index greater than the dimension length! {input.shape[dim]}"
         )
-    output_shape = list(input.shape)
-    output_shape[dim] = 1
-    if dynamic_shape > 0:
-        output_shape = get_shape_with_dynamic_shape(
-            ctx, target, source_ir, name, output_shape, input
-        )
+
     index_value = np.array(index, dtype=np.int32)
```

Comment:
Can index itself be dynamic (ITensor)?

Reply:
Based on the [PyTorch docs](https://pytorch.org/docs/stable/generated/torch.select.html) and [the schema](https://github.com/pytorch/pytorch/blob/a408920817ac3852d27fc96ba85109ff4c981d52/aten/src/ATen/native/native_functions.yaml#L4911), the `index` is always an integer and cannot be a list or tuple of integers. The size of the `indices_tensor` created with `index` and `index_value` will always be a scalar. Therefore, it cannot be dynamic.